cnap/core/qat_zip.py

```diff
@@ -0,0 +1,163 @@
+"""A QAT params module.
+
+This module provides the relevant parameters required to call QAT.
+
+Classes:
+    QzDirection: An enum for the type of the Huffman header supported by QATzip.
+    QzPollingMode: An enum for the type of polling mode supported by QATzip.
+"""
+import ctypes
+import logging
+import time
+from core.frame import FrameZipBase
+from core.qat_params import QzSession, QzSessionParamsDeflate
+
+LOG = logging.getLogger(__name__)
+qziplib = ctypes.cdll.LoadLibrary("/usr/local/lib/libqatzip.so")
+
+SW_BACKUP =1
+LAST= 1
+EXPANSION_RATIO = [5, 20, 50, 100]
+
+
+class QATFrameZip(FrameZipBase):
+    """A class that use QAT to accelerate frame compression and decompression.
+
+    Attributes:
+        _session (QzSession): Compress or decompress. 
+        _deflate_params (QzSessionParamsCommon): Dynamic or Static Huffman headers.     
+    """
+
+    def __init__(self,
+                 session: QzSession = QzSession(),
+                 deflate_params: QzSessionParamsDeflate = QzSessionParamsDeflate()):
+        """Initialize a QATFrameZip object.
+
+        This constructor initializes a QATFrameZip object with the given session
+        and common_params.
+
+        Args:
+            session (QzSession): The QATzip Session.
+            deflate_params (QzSessionParamsDeflate): The QATzip session with deflate parameters.
+        """
+        self._session = session
+        self._deflate_params = deflate_params
+        self.init_session()
+        self.set_session()
+
+    def init_session(self) :
+        """Initialize QAT hardware."""
+        ret = qziplib.qzInit(ctypes.byref(self._session), SW_BACKUP)
+        if ret != 0:
+            LOG.error("Failed to initialize the QATzip session: %s", ret)
+            raise NotImplementedError(f"Initialize QATzip session failed with status:{ret}.")
```

Comment:
Is it appropriate to use `NotImplementedError` here? And the raises should also be added to docstring.